### ISSUE TITLE
[data grid] focusElementRef is always undefined, disabling me to imperatively set focus

### ISSUE THREAD

**Issue by hahanein** [mui/mui-x#12761]
### Steps to reproduce

Link to live example: https://stackblitz.com/edit/vitejs-vite-ee42pc?file=src%2FApp.jsx

Steps:
1. Pass "focusElementRef" to component
2. Navigate to a `Switch` celll via keyboard
3. Press any key

### Current behavior

`focusElementRef` does not automatically give focus to the element (`Switch`) in question.

Furthermore, `focusElementRef` is always `undefined` meaning you cannot use it to imperatively set the focus yourself. 

Because of this no key events are ever passed to the `Switch`.

### Expected behavior

There is some way for me to focus the `Switch` component on keyboard navigation and have it receive `keydown` events.

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Used Firefox.  

  System:
    OS: Linux 5.0 undefined
  Binaries:
    Node: 18.18.0 - /usr/local/bin/node
    npm: 10.2.3 - /usr/local/bin/npm
    pnpm: 8.15.3 - /usr/local/bin/pnpm
  Browsers:
    Chrome: Not Found
  npmPackages:
    @emotion/react: ^11.11.4 => 11.11.4 
    @emotion/styled: ^11.11.5 => 11.11.5 
    @mui/base:  5.0.0-beta.40 
    @mui/core-downloads-tracker:  5.15.15 
    @mui/material: ^5.15.15 => 5.15.15 
    @mui/private-theming:  5.15.14 
    @mui/styled-engine:  5.15.14 
    @mui/system:  5.15.15 
    @mui/types:  7.2.14 
    @mui/utils:  5.15.14 
    @mui/x-data-grid:  7.1.1 
    @mui/x-data-grid-pro: ^7.1.1 => 7.1.1 
    @mui/x-license:  7.1.1 
    @types/react: ^18.2.74 => 18.2.74 
    react: ^18.2.0 => 18.2.0 
    react-dom: ^18.2.0 => 18.2.0 
```
</details>

**Search keywords**: datagrid, focusElementRef, cell, focus, ref, undefined

**Search keywords**:

**Comment by cherniavskii** [comment]
Hey @hahanein 
The `focusElementRef` is only passed to the cells in the `actions` column.
Did you try putting the Switch in the `actions` column? Something like this: https://stackblitz.com/edit/vitejs-vite-fe6uez?file=src%2FApp.jsx

**Comment by hahanein** [comment]
Thanks @cherniavskii That makes sense!
